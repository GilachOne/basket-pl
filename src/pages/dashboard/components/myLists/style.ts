import styled from '@emotion/styled';
import { css } from '@emotion/css';

import { kebab } from '../../../../public';

export const CustomSelect = css`
    .react-dropdown-select-dropdown {
        overflow-y: auto;
        overflow-x: auto;
        width: auto;
        margin-left: -120px;
    }
    .react-dropdown-select-content {
        display: none;
    }
`;

export const Main = styled.main`
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: ${props => props.theme.colors.accent.light};
`;

export const ListBox = styled.div`
    width: 95%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin: 10px;
    background: ${props => props.theme.colors.text.main};
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
`;

export const ListLeftDiv = styled.div`
    margin-left: 20px;
    margin-right: 10px;
    align-items: center;
    justify-content: start;
    word-break: break-all;
    flex-grow: 1;
    flex-shrink: 1;
    cursor: pointer;
`;

export const ListRightDiv = styled.div`
    margin-right: 10px;
    flex-grow: 0;
    flex-shrink: 0;
    justify-content: center;
    align-items: center;
    display: flex;
    flex-direction: row;

    & .${CustomSelect} {
        width: 24px;
        height: 24px;
        min-height: 24px;
        min-width: 24px;
        cursor: pointer;
        border-radius: 50%;
        border: 0;
        background-image: url(${kebab});
        font-size: 16px;
        font-weight: 500;
        :focus-within {
            outline: 0;
            box-shadow: none;
        }
    }
`;

export const ListText = styled.p`
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    color: ${props => props.theme.colors.bg.main};
`;

export const ListCounts = styled.p`
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    color: ${props => props.theme.colors.background.main};
    margin-right: 10px;
`;

export const Logo = styled.img`
    margin-top: 64px;
    margin-bottom: 30px;
`;

export const UnderLogoText = styled.p`
    font-family: 'Roboto';
    font-style: normal;
    font-size: 20px;
    color: ${props => props.theme.colors.text.light};
    margin: 0px;
    word-break: break-all;
`;
