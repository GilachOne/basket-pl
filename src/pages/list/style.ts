import styled from '@emotion/styled';
import { Link } from 'react-router-dom';

export const NavLink = styled(Link)`
    width: min-content;
`;
export const Btn = styled.button`
    background: none;
    border: none;
    width: min-content;
`;
export const ArrowLeftWrap = styled.div`
    display: inline-block;
`;
export const RotateWrap = styled.div`
    display: inline-block;
    padding-left: 1.8rem;
`;
export const ShareWrap = styled.div`
    display: inline-block;
    right: 1.2rem;
    position: absolute;
`;
export const Container = styled.div`
    padding: 1.2rem;
    position: relative;
`;
export const Header = styled.div`
    background-color: ${props => props.theme.colors.background.main};
`;
export const ListHeadName = styled.h2`
	margin-bottom: 0.5rem;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 400;
    font-size: 24px;
    color: ${props => props.theme.colors.text.light2};
	margin-right: 4px;
`;
export const Category = styled.button`
    display: flex;
	border: none;
    flex-direction: row;
    height: 48px;
    border-radius: 0.2rem;
    box-shadow: 0px 0.2rem 0.2rem 0px rgba(2, 2, 2, 0.25);
    background-color: ${props => props.color};
	cursor: pointer;
	font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    align-items: center;
    color: ${props => props.theme.colors.text.light2};
	padding-left: 0px;
`;
export const CategoryColor = styled.div`
    width: 1rem;
    height: inherit;
    background-color: ${props => props.color};
    border-bottom-left-radius: 0.2rem;
    border-top-left-radius: 0.2rem;
	margin-right: 4px;
`;
export const Comment = styled.p`
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    text-align: left;
    align-items: center;
    color: ${props => props.theme.colors.accent.main};
    margin-top: 0px;
	margin-bottom: 8px;
`;
export const InputName = styled.div`
    width: 100%;
    position: relative;
    display: block;
    margin-bottom: 0.5rem;
`;

export const AddCategory = styled.button`
    width: 100%;
    height: 52px;
    background-color: ${props => props.theme.colors.accent.light};
    border-radius: 0.2rem;
    border: transparent;
    box-shadow: 0px 0.2rem 0.2rem 0px rgba(2, 2, 2, 0.25);
    margin-bottom: 1rem;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    text-align: center;
    :disabled {
        background-color: ${props => props.theme.colors.accent.light6};
    }
`;
export const InputTextWrap = styled.div`
    position: relative;
    width: 100%;
    display: block;
    margin-bottom: 0.5rem;
`;
export const BtnAdd = styled.button`
    width: 100%;
    height: 52px;
    background-color: ${props => props.theme.colors.brand.main};
    border-radius: 0.2rem;
    border: transparent;
    box-shadow: 0px 0.2rem 0.2rem 0px rgba(2, 2, 2, 0.25);
	margin-top: 0px;
    margin-bottom: 24px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 18px;
    text-align: center;
    :disabled {
        background-color: ${props => props.theme.colors.accent.light6};
    }
`;
export const Mail = styled.div`
    font-size: 0.9rem;
    color: ${props => props.theme.colors.accent.main};
    position: absolute;
    right: 1rem;
    top: 2em;
    background-color: white;
    border: 1px solid ${props => props.theme.colors.accent.main};
    border-radius: 0.5rem;
    box-shadow: 0px 0.2rem 0.2rem 0px rgba(2, 2, 2, 0.25);
    padding: 5px;
    transition: 0.5s;
`;
export const DivError = styled.div`
    font-size: 0.9rem;
    color: ${props => props.theme.colors.accent.main};
    position: absolute;
    left: 1rem;
    top: 2em;
`;
export const ButtonCirclePlus = styled.button`
	width: 32px;
	height: 32px;
	cursor: pointer;
	border-radius: 50%;
	border: none;
	background: ${props => props.theme.colors.background.main};
    color: ${props => props.theme.colors.text.main};
	font-family: 'Roboto';
	font-style: normal;
	font-weight: 300;
	font-size: 32px;
	display: inline-flex;
	align-items: center;
	justify-content: center;
	margin-top: 0.75rem;
	padding-top: 0px;
    padding-bottom: 0px;
`;
export const Ul = styled.ul`
	display: flex;
	margin: 0;
	padding-left: 0;
    flex-wrap: wrap;
	align-items: center;
`;
export const Li = styled.li`
	list-style: none;
`;