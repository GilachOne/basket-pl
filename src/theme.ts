export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
