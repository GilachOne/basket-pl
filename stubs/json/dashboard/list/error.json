{
    "success": false,
    "error": "Список не загрузился"
}