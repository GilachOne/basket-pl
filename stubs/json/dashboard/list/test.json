[
        {
            "id": "uuid1",
            "listName": "Состоялась 94-я церемония вручения премии «Оскар»: награда за лучший фильм присуждена картине...",
            "purchased": 0,
            "total": 5,
            "timeStamp": 1655042751460,
            "userId": "62a31b1b0d78fc3d44f3edd6"
        },
        {
            "id": "uuid2",
            "listName": "Второй список",
            "purchased": 1,
            "total": 5,
            "timeStamp": 1655042751460,
            "userId": "62a31b1b0d78fc3d44f3edd6"
        },
        {
            "id": "uuid3",
            "listName": "Первый список",
            "purchased": 5,
            "total": 5,
            "timeStamp": 1655042751460,
            "userId": "62a31b1b0d78fc3d44f3edd6"
        }
]
 
