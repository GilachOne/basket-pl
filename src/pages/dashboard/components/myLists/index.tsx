import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { getFeatures } from '@ijl/cli';
import Select from 'react-dropdown-select';

import { URLs } from '../../../../__data__/urls';
import { logo } from '../../../../public';
import { FabButton } from '../fab/style';

import {
    Main,
    ListBox,
    ListLeftDiv,
    ListText,
    ListRightDiv,
    ListCounts,
    UnderLogoText,
    Logo,
    CustomSelect
} from './style';

type MyListProps = {
    lists: {
        id: string;
        listName: string;
        purchased: number;
        total: number;
    }[];
    onRemove: (id: string) => void;
    onRename: (id: string) => void;
    onCopy: (id: string) => void;
    onShare: (url: string) => void;
    onAdd: () => void;
};

const copyFeature = getFeatures('basket')['dashboard.copylist'];

export const MyListsComponent: React.FC<MyListProps> = ({
    lists,
    onRemove,
    onRename,
    onCopy,
    onShare,
    onAdd,
}) => {
    const { t } = useTranslation();

    const calcOptions = useMemo(() => {
        const options = [
            {
                value: 'Rename',
                label: t('basket.dashboard.dropdownmenu.rename'),
            },
            { value: 'Share', label: t('basket.dashboard.dropdownmenu.share') },
        ];
        if (copyFeature) {
        options.push({
            value: 'Copy',
            label: t('basket.dashboard.dropdownmenu.copy'),
        });
        }
        options.push({
            value: 'Delete',
            label: t('basket.dashboard.dropdownmenu.delete'),
        });

        return options;
    }, [t]);

    const onSelect = id => options => {
        if (options[0].value == 'Delete') onRemove(id);
        if (options[0].value == 'Rename') onRename(id);
        if (options[0].value == 'Copy') onCopy(id);
        if (options[0].value == 'Share') onShare(URLs.list.getUrl(id));
    };

    const navigate = useNavigate();

    const navigateList = id => {
        return navigate(URLs.list.getUrl(id));
    };

    return (
        <Main>
            {lists?.map(listItem => (
                <ListBox key={listItem.id}>
                    <ListLeftDiv
                        onClick={() => navigateList(listItem.id)}
                        data-testid={listItem.id}
                    >
                        <ListText>{listItem.listName}</ListText>
                    </ListLeftDiv>
                    <ListRightDiv>
                        <ListCounts>{`${listItem.purchased} / ${listItem.total}`}</ListCounts>
                        <Select
                            className={CustomSelect}
                            options={calcOptions}
                            onChange={onSelect(listItem.id)}
                            values={[]}
                            placeholder=""
                            dropdownHandle={false}
                        />
                    </ListRightDiv>
                </ListBox>
            ))}
            {(!lists || !lists.length) && (
                <>
                    <Logo src={logo} alt="logo" />
                    <UnderLogoText>
                        {t('basket.dashboard.emptyList.text1')}
                    </UnderLogoText>
                    <UnderLogoText>
                        {t('basket.dashboard.emptyList.text2')}
                    </UnderLogoText>
                </>
            )}
            <FabButton onClick={onAdd} data-testid="addButton">
                +
            </FabButton>
        </Main>
    );
};
