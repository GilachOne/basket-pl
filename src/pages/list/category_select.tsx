import React, { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { Form as FinalForm, Field } from 'react-final-form';
import { useSelector, useDispatch } from 'react-redux';

import {
    Container,
    ListHeadName,
    Comment,
    InputName,
    AddCategory,
    InputTextWrap,
    BtnAdd,
	ButtonCirclePlus,
	Ul,
	Li,
} from '../list/style';

import { Header2 } from './components/headers/headers';
import { CategoryItem } from './components/category';
import { Input } from '../../components/category/input';
import { InputColor } from '../../components/category/select';
import { Textarea } from '../../components/category/textarea';
import { Categories } from '../../components/category/div';
import {
    useCategoriesAddMutation,
    useCategoriesCurrentQuery,
    useAddItemMutation,
} from '../../__data__/api/main';
import { Loader } from '../../components/loader';
import { SnackBar } from '../dashboard/components/snackbar/style';
import { ErrorSpan } from '../../components/error/span';
import { actions } from '../../__data__/slices/errorTextSlice';
import { RootState } from '../../__data__/store';
import { useNavigate, useParams } from 'react-router-dom';
import { URLs } from '../../__data__/urls';

const CategorySelect = () => {
    const { t } = useTranslation();
    const param = useParams();
    const snackError = useSelector((state: RootState) => state.errorText.error);
    const dispatch = useDispatch();
    const navigate = useNavigate();
    const {
        data: getCategoriesData,
        isError: getCategoriesIsError,
        error: getCategoriesError,
        isLoading: getCategoriesLoading,
        isSuccess: getCategoriesSuccess,
    } = useCategoriesCurrentQuery();
    const [
        addCategoryData,
        {
            isLoading: postCategryLoading,
            data: postCategoryData,
            isError: postCategoryIsError,
            error: postCategoryError,
            isSuccess: postCategorySuccess,
        },
    ] = useCategoriesAddMutation();

    const [
        addItemData,
        {
            isLoading: postItemLoading,
            data: postItemData,
            isError: postItemIsError,
            error: postItemError,
            isSuccess: postItemSuccess,
        },
    ] = useAddItemMutation();

	const [addCategory, setAddCategory] = useState(false);
	const addButtonCategory = () => {
		setAddCategory(true);
	}
	
	const onSubmitAddCategory = async data => {
		await addCategoryData(data);
	};

    useEffect(() => {
        const errorCategoriesText = getCategoriesError?.data?.error;

        if (getCategoriesIsError) {
            dispatch(
                actions.setError(
                    errorCategoriesText || t('basket.add.default.error.text')
                )
            );
            setTimeout(() => {
                dispatch(actions.resetError());
            }, 3800);
        }
        if (postItemSuccess) {
            navigate(URLs.list.getUrl(param.id));
        }
		if (postCategorySuccess) {
			setAddCategory(false);
		}
    }, [
        dispatch,
        getCategoriesError,
        getCategoriesIsError,
        postCategoryData,
        navigate,
        postItemSuccess,
        t,
        param,
		postCategorySuccess
    ]);

    const [categorySelected, setCategorySelected] = useState<string>();


    const onSubmitAddItem = data => {
        addItemData({ itemDataAdd: data, listId: param.id });
    };

    const createRequiredFieldValidator = errorMessage => value =>
        value ? undefined : errorMessage;
	
	const createСheckFieldNameItem = errorMessage => value =>
        /(^[а-яА-ЯёЁa-zA-Z0-9-,%№ ]+$)/.test(value) ? undefined : errorMessage;
	
    const createСheckFieldNameCategory = errorMessage => value =>
		/(^[а-яА-ЯёЁa-zA-Z- ]+$)/.test(value) ? undefined : errorMessage;

    const composeValidators =
        (...validators) =>
        value =>
            validators.reduce(
                (error, validator) => error || validator(value),
                undefined
            );

    const [untouchedProductName, setUntouchedProductName] = useState(false);
    const actionProduct = () => {
        setUntouchedProductName(true);
    };

    return (
        <div>
            <Header2 />
            {snackError && <SnackBar error>{snackError}</SnackBar>}
            {!getCategoriesLoading && !getCategoriesIsError && (
                <Container>
                    <Ul>
						<Li>
							<ListHeadName>
								{t('basket.add.list.head.name')}
							</ListHeadName>
						</Li>
						<Li>
							{!addCategory && <ButtonCirclePlus type="button" title={t('basket.add.button.plus')} onClick={addButtonCategory}>+</ButtonCirclePlus>}
						</Li>
					</Ul>
					
                    <FinalForm
                        onSubmit={onSubmitAddItem}
                        initialValues={{ categoryId: categorySelected }}
                        keepDirtyOnReinitialize={true}
                        render={({ handleSubmit }) => (
                            <form onSubmit={handleSubmit}>
                                <Field
                                    name="categoryId"
                                    validate={composeValidators(
                                        createRequiredFieldValidator(
                                            t(
                                                'basket.add.require.product.category.name'
                                            )
                                        )
                                    )}
                                    render={({ meta }) =>
                                        getCategoriesSuccess && (
                                            <Categories
                                                error={
                                                    untouchedProductName &&
                                                    !categorySelected &&
                                                    meta.error
                                                }
                                            >
                                                {getCategoriesData?.data?.map(
                                                    categ => (
                                                        <CategoryItem
                                                            key={categ.id}
                                                            category={categ}
                                                            onClick={
                                                                setCategorySelected
                                                            }
                                                            isPressed={
                                                                categ.id ===
                                                                categorySelected
                                                            }
                                                        />
                                                    )
                                                )}
                                            </Categories>
                                        )
                                    }
                                />
                                <Comment>
                                    {t('basket.add.list.product.name')}
                                </Comment>
                                <InputTextWrap>
                                    <Field
                                        name="text"
                                        validate={composeValidators(
                                            createRequiredFieldValidator(
                                                t(
                                                    'basket.add.require.product.name'
                                                )
                                            ),
                                            createСheckFieldNameItem(
                                                t(
                                                    'basket.add.check.product.name'
                                                )
                                            )
                                        )}
                                        render={({ input, meta }) => (
                                            <Textarea
                                                {...input}
                                                error={
                                                    meta.touched && meta.error
                                                }
                                                placeholder={t(
                                                    'basket.add.input.text'
                                                )}
                                                onFocus={actionProduct}
                                                disabled={postItemLoading}
                                            />
                                        )}
                                    />
                                </InputTextWrap>
                                <BtnAdd
                                    type="submit"
                                    title={t('basket.add.button.item')}
                                    disabled={postItemLoading}
                                >
                                    {t('basket.add.button.item')}
                                </BtnAdd>
                                {postItemIsError && (
                                    <ErrorSpan>
                                        {postItemError?.data?.error ||
                                            t('backet.add.item.error.text')}
                                    </ErrorSpan>
                                )}
                            </form>
                        )}
                    />
					{addCategory && 
						<FinalForm
							onSubmit={onSubmitAddCategory}
							render={({ handleSubmit, form: { restart } }) => (
								<form
									onSubmit={event =>
										handleSubmit(event)?.then(() => restart())
									}
								>
									<Comment>{t('basket.add.comment')}</Comment>
									<InputName>
										<Field
											name="name"
											validate={composeValidators(
												createRequiredFieldValidator(
													t(
														'basket.add.require.category.name'
													)
												),
												createСheckFieldNameCategory(
													t(
														'basket.add.check.category.name'
													)
												)
											)}
											render={({ input, meta }) => (
												<Input
													type="text"
													disabled={postCategryLoading}
													{...input}
													error={
														meta.touched && meta.error
													}
													placeholder={t(
														'basket.add.input.name.text'
													)}
												/>
											)}
										/>
										<Field
											name="color"
											validate={composeValidators(
												createRequiredFieldValidator(
													t(
														'basket.add.require.category.color'
													)
												)
											)}
											render={({ input, meta }) => (
												<InputColor
													data-testid="select"
													disabled={postCategryLoading}
													{...input}
													error={
														meta.touched && meta.error
													}
												>
													<option value="" disabled>
														{t(
															'basket.add.input.color'
														)}
													</option>
													<option value="#08AE0F">
														{t(
															'basket.add.input.color.green'
														)}
													</option>
													<option value="#9D79B9">
														{t(
															'basket.add.input.color.purple'
														)}
													</option>
													<option value="#B11F1F">
														{t(
															'basket.add.input.color.scarlet'
														)}
													</option>
													<option value="#3414F5">
														{t(
															'basket.add.input.color.blue'
														)}
													</option>
													<option value="#008060">
														{t(
															'basket.add.input.color.darkGreen'
														)}
													</option>
													<option value="#D928D2">
														{t(
															'basket.add.input.color.pink'
														)}
													</option>
													<option value="#9B763D">
														{t(
															'basket.add.input.color.brown'
														)}
													</option>
													<option value="#830DF9">
														{t(
															'basket.add.input.color.violet'
														)}
													</option>
													<option value="#FA8803">
														{t(
															'basket.add.input.color.orange'
														)}
													</option>
													<option value="#FCF20E">
														{t(
															'basket.add.input.color.yellow'
														)}
													</option>
													<option value="#EC1818">
														{t(
															'basket.add.input.color.red'
														)}
													</option>
													<option value="#14F52B">
														{t(
															'basket.add.input.color.lightGreen'
														)}
													</option>
												</InputColor>
											)}
										/>
									</InputName>
									<AddCategory
										type="submit"
										title={t('basket.add.button.category')}
										disabled={postCategryLoading}
									>
										{t('basket.add.button.category')}
									</AddCategory>
									{postCategoryIsError && (
										<ErrorSpan>
											{postCategoryError?.data?.error ||
												t('backet.add.category.error.text')}
										</ErrorSpan>
									)}
								</form>
							)}
						/>
					}
                </Container>
            )}
            {getCategoriesLoading && <Loader />}
        </div>
    );
};

export default CategorySelect;
