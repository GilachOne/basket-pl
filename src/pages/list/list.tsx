import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import {
    useGetShoppingListQuery,
    useDashboardListQuery,
} from '../../__data__/api/main';
import { URLs } from '../../__data__/urls';
import { Loader } from '../../components/loader';
import { FabButton } from '../dashboard/components/fab/style';
import { Header1 } from './components/headers/headers';
import { Item } from './components/item/item';

import { Container, ListHeadName } from './style';

const List = () => {
    const { t } = useTranslation();
    const param = useParams();

    const {
        data: itemsRespons,
        isLoading,
        isSuccess,
    } = useGetShoppingListQuery(param.id);
    const { data: dashboardResp } = useDashboardListQuery();

    const list = dashboardResp?.data.find(list => list.id === param.id);

    return (
        <div>
            <Header1 />
            <Container>
                <div>
                    <ListHeadName>
                        {(dashboardResp && list?.listName) || 'Чужой список'}
                    </ListHeadName>
                    {isSuccess &&
                        itemsRespons.data?.map(item => (
                            <Item item={item} key={item.id} {...item} />
                        ))}
                    <Link
                        to={URLs.add.getUrl(param.id)}
                        title={t('basket.list.add')}
                    >
                        <FabButton>+</FabButton>
                    </Link>
                </div>
                {isLoading && <Loader />}
            </Container>
        </div>
    );
};

export default List;
