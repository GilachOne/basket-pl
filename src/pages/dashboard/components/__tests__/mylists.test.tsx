import React from 'react';
import { describe, it, expect, jest } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { BrowserRouter } from 'react-router-dom';

import { MyListsComponent } from '../myLists';
import { theme } from '../../../../theme';
import list from '../../../../../stubs/json/dashboard/list/test.json';

describe('MyListsComponent', () => {
    it('MyListsComponent Empty', () => {
        const onAdd = jest.fn();
        const onRemove = jest.fn();
        const onRename = jest.fn();
        const onCopy = jest.fn();
        const onShare = jest.fn();

        render(
            <ThemeProvider theme={theme}>
                <BrowserRouter>
                    <MyListsComponent
                        lists={null}
                        onAdd={onAdd}
                        onRemove={onRemove}
                        onRename={onRename}
                        onCopy={onCopy}
                        onShare={onShare}
                    />
                </BrowserRouter>
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTestId('addButton'));

        expect(onAdd).toBeCalled();
        expect(document.body).toMatchSnapshot();
    });

    it('MyListsComponent With list', () => {
        const onAdd = jest.fn();
        const onRemove = jest.fn();
        const onRename = jest.fn();
        const onCopy = jest.fn();
        const onShare = jest.fn();

        render(
            <ThemeProvider theme={theme}>
                <BrowserRouter>
                    <MyListsComponent
                        lists={list}
                        onAdd={onAdd}
                        onRemove={onRemove}
                        onRename={onRename}
                        onCopy={onCopy}
                        onShare={onShare}
                    />
                </BrowserRouter>
            </ThemeProvider>
        );

        fireEvent.click(screen.getByTestId('uuid1'));
        fireEvent.click(screen.getByTestId('addButton'));

        expect(onAdd).toBeCalled();
        expect(document.body).toMatchSnapshot();
    });
});
