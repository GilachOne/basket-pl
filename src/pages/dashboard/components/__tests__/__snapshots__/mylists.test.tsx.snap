// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListsComponent MyListsComponent Empty 1`] = `
<body>
  <div>
    <main
      class="css-1jsn5n8"
    >
      <img
        alt="logo"
        class="css-wnwrug"
        src="logo.png"
      />
      <p
        class="css-f4jk5p"
      >
        Создайте свой первый список.
      </p>
      <p
        class="css-f4jk5p"
      >
        Сделайте покупки быстрее и удобнее
      </p>
      <button
        class="css-ftp01x"
        data-testid="addButton"
      >
        +
      </button>
    </main>
  </div>
</body>
`;

exports[`MyListsComponent MyListsComponent With list 1`] = `
<body>
  <div>
    <main
      class="css-1jsn5n8"
    >
      <div
        class="css-qte5c2"
      >
        <div
          class="css-1nclwr3"
          data-testid="uuid1"
        >
          <p
            class="css-14z2g2o"
          >
            Состоялась 94-я церемония вручения премии «Оскар»: награда за лучший фильм присуждена картине...
          </p>
        </div>
        <div
          class="css-115o3ib"
        >
          <p
            class="css-9x7icz"
          >
            0 / 5
          </p>
          <div>
            <div
              aria-expanded="false"
              aria-label="Dropdown select"
              class="react-dropdown-select css-14g55hk css-16vukcu e1gzf2xs0"
              color="#0074D9"
              direction="ltr"
              tabindex="0"
            >
              <div
                class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
              >
                <input
                  class="react-dropdown-select-input css-1ov1ozn e11wid6y0"
                  placeholder=""
                  tabindex="-1"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-qte5c2"
      >
        <div
          class="css-1nclwr3"
          data-testid="uuid2"
        >
          <p
            class="css-14z2g2o"
          >
            Второй список
          </p>
        </div>
        <div
          class="css-115o3ib"
        >
          <p
            class="css-9x7icz"
          >
            1 / 5
          </p>
          <div>
            <div
              aria-expanded="false"
              aria-label="Dropdown select"
              class="react-dropdown-select css-14g55hk css-16vukcu e1gzf2xs0"
              color="#0074D9"
              direction="ltr"
              tabindex="0"
            >
              <div
                class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
              >
                <input
                  class="react-dropdown-select-input css-1ov1ozn e11wid6y0"
                  placeholder=""
                  tabindex="-1"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-qte5c2"
      >
        <div
          class="css-1nclwr3"
          data-testid="uuid3"
        >
          <p
            class="css-14z2g2o"
          >
            Первый список
          </p>
        </div>
        <div
          class="css-115o3ib"
        >
          <p
            class="css-9x7icz"
          >
            5 / 5
          </p>
          <div>
            <div
              aria-expanded="false"
              aria-label="Dropdown select"
              class="react-dropdown-select css-14g55hk css-16vukcu e1gzf2xs0"
              color="#0074D9"
              direction="ltr"
              tabindex="0"
            >
              <div
                class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
              >
                <input
                  class="react-dropdown-select-input css-1ov1ozn e11wid6y0"
                  placeholder=""
                  tabindex="-1"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="css-ftp01x"
        data-testid="addButton"
      >
        +
      </button>
    </main>
  </div>
</body>
`;
